File: delegapter/src/main/java/net/aquadc/delegapter/RrAL.java
package net.aquadc.delegapter;

import java.util.ArrayList;
import java.util.Collection;

/**
 * ArrayList with public removeRange().
 * @author Mike Gorünóv
 */
public class RrAL<E> extends ArrayList<E> {
    private RrAL() { super(); }
    private RrAL(int initialCapacity) { super(initialCapacity); }
    public /*called from .adapter package*/ RrAL(Collection<? extends E> copyFrom) { super(copyFrom); }
    static <E> RrAL<E> create(int initialCapacity) {
        return initialCapacity < 0 ? new RrAL<>() : new RrAL<>(initialCapacity);
    }

    @Override public void removeRange(int fromIndex, int toIndex) {
        super.removeRange(fromIndex, toIndex);
    }
}

File: delegapter/src/main/java/net/aquadc/delegapter/RrAL.kt
package net.aquadc.delegapter

/**
 * ArrayList with public removeRange().
 * @author Mike Gorünóv
 */
class RrAL<E> : ArrayList<E> {
    private constructor() : super()
    private constructor(initialCapacity: Int) : super(initialCapacity)
    internal constructor(copyFrom: Collection<E>?) : super(copyFrom)

    public override fun removeRange(fromIndex: Int, toIndex: Int) {
        super.removeRange(fromIndex, toIndex)
    }

    internal companion object {
        fun <E> create(initialCapacity: Int): RrAL<E> {
            return if (initialCapacity < 0) RrAL() else RrAL(initialCapacity)
        }
    }
}
